### PR TITLE
Add Yubikey OTP Auth

### PR DESCRIPTION
**Pull request by di4bl0te**
Add Yubikey OTP Auth

### CODE REVIEW

File: src/install/hashtopussy.sql [hunk 1]
```diff
@@ -560,7 +563,12 @@ CREATE TABLE `User` (
   `lastLoginDate`      INT(11)          NOT NULL,
   `registeredSince`    INT(11)          NOT NULL,
   `sessionLifetime`    INT(11)          NOT NULL DEFAULT '600',
-  `rightGroupId`       INT(11)          NOT NULL DEFAULT '1'
+  `rightGroupId`       INT(11)          NOT NULL DEFAULT '1',
+  `yubikey`            INT(1)           NOT NULL DEFAULT '0',
+  `otp1`               VARCHAR(50)      NOT NULL,
```
Comment (s3inlc) [inline src/install/hashtopussy.sql]: Is it correct that otp1 is varchar and the other three are ints?
As what I see from the other code, they should all have the same type.

File: src/index.php [hunk 2]
```diff
@@ -29,6 +29,8 @@
         break;
       case '4':
         $message = "<div class='alert alert-warning'>You need to be logged in to view this! Please log in again.</div>";
+      case '':
```
Comment (s3inlc) [inline src/index.php]: Why the case here is empty?
As there is no break statement after the other case the OTP error message always will pop up when someone gets logged out. 
As the login procedure currently doesn't report why the login failed, maybe just include this error message into the failed login attempt (case 3).

File: src/inc/db.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (s3inlc) [inline src/inc/db.php]: db.php should not exist in the repository as it should be created from db.template.php either manually or by the installation process.

File: src/install/index.php [hunk 4]
```diff
@@ -17,6 +17,7 @@
 use DBA\User;
 
 require_once(dirname(__FILE__) . "/../inc/load.php");
+require_once(dirname(__FILE__) . "/../inc/db.php");
```
Comment (s3inlc) [inline src/install/index.php]: The db files already get included by load.php, so this require is not necessary

File: src/install/hashtopussy.sql (hunk 1 above)
Comment (djban) [inline src/install/hashtopussy.sql]: VARCHAR es el correcto para los 4 otp

File: src/inc/db.php (hunk 3 above)
Comment (djban) [inline src/inc/db.php]: ok, lo borro.

File: src/install/hashtopussy.sql (hunk 1 above)
Comment (djban) [inline src/install/hashtopussy.sql]: Tengo que modificar varias cosas, no lo envies aun.

Comment (djban) [inline src/install/hashtopussy.sql]: Terminado @s3inlc.

**Comment by s3inlc** [comment]
as mentioned in the referencing pull request, changes are included so this pull request is obsolete.

**Comment by winxp5421** [comment]
Thank you @di4bl0te for your contribution.

**Pull request closed without merging**